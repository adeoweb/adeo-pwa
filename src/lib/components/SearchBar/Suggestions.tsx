import React, { FunctionComponent } from 'react';
import SuggestedProducts from './SuggestedProducts';
import { TProductsSearch } from 'src/lib/types/ProductSearch';
import { useSuggestions } from 'src/peregrine/lib/talons/adeoweb/SearchBar/useSuggestions';

type TSuggestionsProps = {
    products?: TProductsSearch;
    setVisible: (isVisible: boolean) => void;
    visible: boolean;
};

const Suggestions: FunctionComponent<TSuggestionsProps> = ({
    products,
    setVisible,
    visible
}) => {
    const items = products && products.items ? products.items : [];

    const { onNavigate, shouldRender } = useSuggestions({
        items,
        setVisible,
        visible
    });

    if (!shouldRender) {
        return null;
    }

    return <SuggestedProducts onNavigate={onNavigate} products={items} />;
};

export default Suggestions;
