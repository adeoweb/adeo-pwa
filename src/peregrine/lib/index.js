// components
export { default as PeregrineContextProvider } from './PeregrineContextProvider';
