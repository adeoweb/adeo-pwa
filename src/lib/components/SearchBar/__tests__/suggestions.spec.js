import React from 'react';

import { createTestInstance } from '@magento/peregrine';

import Suggestions from '../Suggestions';

jest.mock('../SuggestedProducts', () => () => null);

jest.mock('src/peregrine/lib/talons/adeoweb/SearchBar/useSuggestions', () => {
    return {
        useSuggestions: () => ({
            onNavigate: jest.fn(),
            shouldRender: true
        })
    };
});
const setVisible = jest.fn();
const products = {};

test('renders correctly when visible', () => {
    const instance = createTestInstance(
        <Suggestions
            products={products}
            visible={true}
            setVisible={setVisible}
        />
    );

    expect(instance.toJSON()).toMatchSnapshot();
});
