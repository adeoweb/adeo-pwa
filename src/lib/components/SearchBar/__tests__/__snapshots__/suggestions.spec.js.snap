// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when visible 1`] = `null`;
